File: backend/src/main/java/com/examreg/examreg/dto/response/AuthResponse.java
package com.examreg.examreg.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class AuthResponse<T> {
  private String token;
  private String refreshToken;
  private T user; 
}

File: frontend/src/models/AuthResponse.js
import { User } from "./User";

export class AuthResponse {
  constructor(data) {
    this.message = data.message;
    this.token = data.data.token;
    this.refreshToken = data?.data?.refreshToken ?? null;
    this.user = new User(data.data.user)
  }

  hasToken() {
    return !!this.token;
  }
  
  isFirstLogin() {
    return this.user.firstLogin === true;
  }
  
  static fromJSON(json) {
    return new AuthResponse(json);
  }
}